nengo_spinnaker/operators/filter.py

```diff
@@ -1,111 +1,325 @@
+                self.output_keys_region.write_subregion_to_file(
+                    region_mem[1], vertex.out_slice, cluster=vertex.cluster)
+
+                # Input filters
+                self.filters_region.write_subregion_to_file(region_mem[2])
+
+                # Filter routing
+                self.routing_region.write_subregion_to_file(region_mem[3])
+
+                # Transform matrix
+                self.transform_region.write_subregion_to_file(
+                    region_mem[4], vertex.out_slice)
+
+
+class ParallelFilterSlice(Vertex):
+    """Represents a portion of a parallel filter.
+
+    Attributes
+    ----------
+    in_slice : :py:class:`slice`
+        Slice of the input space managed by this vertex.
+    out_slice : :py:class:`slice`
+        Slice of the output space managed by this vertex.
+    """
+    def __init__(self, in_slice, out_slice, resources=dict(),
+                 transmission_parameter_slices=list()):
+        super(ParallelFilterSlice, self).__init__(get_application("filter"),
+                                                  resources)
+
+        # Store the slices
+        self.in_slice = in_slice
+        self.out_slice = out_slice
+
+        # Store which signal parameter slices we contain
+        self.transmission_params = list()
+        out_set = set(range(out_slice.start or 0,
+                            out_slice.stop or 0,
+                            out_slice.step or 1))
+        for transmission_params, outs in transmission_parameter_slices:
+            # If there is an intersection between the outs and the set of outs
+            # we're responsible for then store transmission parameters.
+            if out_set & outs:
+                self.transmission_params.append(transmission_params)
+
+    def accepts_signal(self, signal_params, transmission_params):
+        """Choose whether to receive this signal or not."""
+        if isinstance(transmission_params, EnsembleTransmissionParameters):
+            # If the connection is from an ensemble only return true if the
+            # decoders contain non-zero values in the input dimensions we care
+            # about.
+            return np.any(transmission_params.decoders[:, self.in_slice])
+        elif isinstance(transmission_params,
+                        PassthroughNodeTransmissionParameters):
+            # If the connection is from a Node of some variety then only return
+            # true if the transform contains non-zero values in the rows which
+            # relate to the subspace we receive input in.
+            return np.any(transmission_params.transform[self.in_slice])
+
+        # We don't know how to interpret the transmission parameters
+        raise NotImplementedError
+
+    def transmits_signal(self, signal_params, transmission_params):
+        """Choose whether we transmit this signal or not."""
+        return transmission_params in self.transmission_params
+
+
+class SystemRegion(object):
```

Comment:
I was planning on something like that, what is currently the system region in most objects should be split into "SystemRegion", which will be shared, and "SpecificThingRegion".